fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java

```diff
@@ -146,6 +149,37 @@ public boolean hasPermission(
     return false;
   }
 
+  public boolean canCreate(String resourceType, Object resource) {
+    if (!fiatStatus.isEnabled()) {
+      return true;
+    }
+
+    String username = getUsername(SecurityContextHolder.getContext().getAuthentication());
+
+    try {
+      return AuthenticatedRequest.propagate(
+              () -> {
+                return retryHandler.retry(
+                    "determine whether " + username + " can create resource" + resource,
+                    () -> {
+                      Response response = fiatService.canCreate(username, resourceType, resource);
```

Comment:
The response handling bit in here is incorrect.

On a 4xx response from the server, retrofit throws a RetrofitException

This should looks something like:

```suggestion
                      try {
                        fiatService.canCreate(username, resourceType, resource);
                        return true;
                      } catch (RetrofitError re) {
                        boolean shouldRetry = true;
                        if (re.getKind() == RetrofitError.Kind.HTTP) {
                          switch (HttpStatus.valueOf(re.getResponse().getStatus())) {
                            case NOT_FOUND:
                              return false;

                            case BAD_REQUEST:
                              shouldRetry = false;
                              break;
                          }
                        }
                        IntegrationException ie = new IntegrationException(re);
                        ie.setRetryable(shouldRetry);
                        throw ie;
                      }
```

Reply:
The retry handler doesn't look like it cares about the `retryable` property. Or is this for the future?